### PR TITLE
Adding OFED blacklist management to MOFED container

### PR DESCRIPTION
**Pull request by heyvister1**
Adding OFED blacklist management to be handeled by MOFED container instead of network-operator.

* blacklist modules will be done only during openibd restart
Signed-off-by: Ido Heyvi <iheyvi@nvidia.com>

```
[07-Nov-24_06:29:42] Apply blacklisted mofed modules file to host (/etc/modprobe.d/blacklist-ofed-modules.conf)
Function: generate_ofed_modules_blacklist
Unloading HCA driver:                                      [  OK  ]
Loading HCA driver and Access Layer:                       [  OK  ]
[07-Nov-24_06:30:04] Remove blacklisted mofed modules file from host
[07-Nov-24_06:30:04] Restoring SR-IOV config for device: 0000:03:00.0 (VFs=8) netdev name: ibp3s0f0
```

### CODE REVIEW

**Review by rollandf — APPROVED** [review]

File: entrypoint.sh [hunk 1]
```diff
@@ -467,9 +485,13 @@ function restart_driver() {
 
     ${load_pci_hyperv_intf} && exec_cmd "modprobe -d /host pci-hyperv-intf"
 
+    timestamp_print "Apply blacklisted mofed modules file to host (${OFED_BLACKLIST_MODULES_FILE})"
+    generate_ofed_modules_blacklist 
     ${UNLOAD_STORAGE_MODULES} && unload_storage_modules
 
     exec_cmd "/etc/init.d/openibd restart"
+    timestamp_print "Remove blacklisted mofed modules file from host"
```
Comment (e0ne) [inline entrypoint.sh:493]: Based on [1] and [2] this code won't be executed if openibd restart fails

[1] https://github.com/Mellanox/doca-driver-build/blob/69d08d646db208b5e1e16b489f831b0d239986fe/entrypoint.sh#L153
[2] https://github.com/Mellanox/doca-driver-build/blob/69d08d646db208b5e1e16b489f831b0d239986fe/entrypoint.sh#L62

Comment (heyvister1) [reply entrypoint.sh]: Fixed by using trap

File: entrypoint.sh [hunk 2]
```diff
@@ -467,9 +489,14 @@ function restart_driver() {
 
     ${load_pci_hyperv_intf} && exec_cmd "modprobe -d /host pci-hyperv-intf"
 
+    timestamp_print "Apply blacklisted mofed modules file to host (${OFED_BLACKLIST_MODULES_FILE})"
+    trap 'remove_ofed_modules_blacklist' EXIT
+    generate_ofed_modules_blacklist 
     ${UNLOAD_STORAGE_MODULES} && unload_storage_modules
 
     exec_cmd "/etc/init.d/openibd restart"
+    remove_ofed_modules_blacklist
```
Comment (rollandf) [inline entrypoint.sh:498]: I think we should also call to `remove_ofed_modules_blacklist` in `exit_entryp()` to make sure it is cleaned

Comment (heyvister1) [reply entrypoint.sh:498]: Why do you think that file worn't be removed?

Comment (e0ne) [reply entrypoint.sh:498]: `exit_entryp()` exits with error code if `exec_cmd` failed so this line won't be executed on a failure

Comment (heyvister1) [reply entrypoint.sh:498]: I understand, but for error we have the trap calll on line [L493](https://github.com/Mellanox/doca-driver-build/pull/8/files#diff-6f9d41d046756f0ddc2fcee0626bdb50100d12b88f293734eff742818e03efa2R493)

Comment (rollandf) [reply entrypoint.sh:498]: I see it now, wasn't there when I reviewed.

**Review by e0ne — APPROVED** [review]
Thanks for addressing my comments!

**Pull request merged**